tests/pytests/integration/modules/test_mac_shadow.py

```diff
@@ -0,0 +1,226 @@
+"""
+integration tests for mac_shadow
+"""
+
+import datetime
+
+import pytest
+from saltfactories.utils import random_string
+
+pytestmark = [
+    pytest.mark.skip_if_binaries_missing("dscl", "pwpolicy"),
+    pytest.mark.skip_initial_gh_actions_failure,
+    pytest.mark.slow_test,
+    pytest.mark.skip_if_not_root,
+    pytest.mark.skip_unless_on_darwin,
+]
+
+
+@pytest.fixture(scope="function")
+def setup_teardown_vars(salt_call_cli):
+    TEST_USER = random_string("RS-", lowercase=False)
+    NO_USER = random_string("RS-", lowercase=False)
+
+    salt_call_cli.run("user.add", TEST_USER)
+
+    try:
+        yield TEST_USER, NO_USER
+    finally:
+        salt_call_cli.run("user.delete", TEST_USER)
+
+
+def test_info(salt_call_cli, setup_teardown_vars):
+    """
+    Test shadow.info
+    """
+    TEST_USER = setup_teardown_vars[0]
+    NO_USER = setup_teardown_vars[1]
+
+    # Correct Functionality
+    ret = salt_call_cli.run("shadow.info", TEST_USER)
+    assert ret.data["name"] == TEST_USER
+
+    # User does not exist
+    ret = salt_call_cli.run("shadow.info", NO_USER)
+    assert ret.data["name"] == ""
```

Comment:
```suggestion
def test_info(salt_call_cli, accounts):
    """
    Test shadow.info
    """
    TEST_USER = setup_teardown_vars[0]
    NO_USER = setup_teardown_vars[1]

    # Correct Functionality
    ret = salt_call_cli.run("shadow.info", accounts.created)
    assert ret.data["name"] == accounts.created

    # User does not exist
    ret = salt_call_cli.run("shadow.info", accounts.not_created)
    assert ret.data["name"] == ""
```

Same for the remaining tests....